File: javascript/phone-number/phone-number.js

export class PhoneNumber {
  constructor(number) {
    const cleanedNumbers = number
      .replace(/\D/g, '')
      .replace(/^1/, '');
    this._number = /^[2-9]\d{2}[2-9]\d{6}$/.test(cleanedNumbers)
      ? cleanedNumbers
      : null;
  }

  number() {
    return this._number;
  }
}

File: kotlin/phone-number/src/main/kotlin/PhoneNumber.kt
class PhoneNumber(rawNumber: String) {

    val number: String

    init {
        val numbers = rawNumber.replace("\\D".toRegex(), "")
        val validNumbersRegex = """^1?[2-9]\d\d[2-9]\d{6}$""".toRegex()

        require(numbers.matches(validNumbersRegex))

        number = if (numbers.length == 11) numbers.drop(1) else numbers
    }

}
